### PR TITLE
Adding support of UTF8 file name processing in mio on windows

### PR DESCRIPTION
**Pull request by shivendra14**
Contributing back changes from app.

### CODE REVIEW

File: single_include/mio/mio.hpp [hunk 1]
```diff
@@ -793,6 +793,16 @@ inline DWORD int64_low(int64_t n) noexcept
...
     return n & 0xffffffff;
 }
+	
+std::wstring s2ws(const std::string& s)
+{
+    if (s.empty()) 
+        return{};
+    const auto sLength = static_cast<int>(s.length());
+    auto buf = std::vector<wchar_t>(sLength);
+    const auto wideCharCount = MultiByteToWideChar(CP_UTF8, 0, s.c_str(), sLength, buf.data(), sLength);
```
Comment (mandreyel) [inline single_include/mio/mio.hpp:803]: nitpick: Can the variables here use camel case to be consistent with the rest of the codebase (widnows conventions notwithstanding)?

**Comment by mandreyel** [comment]
Thank you for this!

One request: could you please move the changes to the following file and then run `python3 third_party/amalgamate.py`? This produces the single header file programmatically.

https://github.com/mandreyel/mio/blob/master/include/mio/detail/mmap.ipp#L70

File: single_include/mio/mio.hpp (hunk 1 above)
Comment (shivendra14) [reply single_include/mio/mio.hpp:803]: you mean the snake case for 
wide_char_count and s_length?

Comment (mandreyel) [reply single_include/mio/mio.hpp:803]: yep

**Pull request closed without merging**